src/midi/midi_fluidsynth.cpp

```diff
@@ -51,13 +52,6 @@ static void init_fluid_dosbox_settings(Section_prop &secprop)
 	        "An optional percentage will scale the SoundFont's volume.\n"
 	        "For example: 'soundfont.sf2 50' will attenuate it by 50 percent.\n"
 	        "The scaling percentage can range from 1 to 500.");
-
-	auto *int_prop = secprop.Add_int("synth_threads", when_idle, 1);
-	int_prop->SetMinMax(1, 256);
-	int_prop->Set_help(
-	        "If set to a value greater than 1, then additional synthesis\n"
-	        "threads will be created to take advantage of many CPU cores.\n"
-	        "(min 1, max 256)");
```

Comment:
Why this value got removed? It's not relevant to change that is being implemented here.

I am not totally *against* this change, but it should probably be done with separate commit with explanation *why* it is being implemented.

Reply:
Right; the PR moves FluidSynth to a dedicated thread which allows FluidSynth's internal synthesizing behavior to proceed in pure linear non-threaded fashion (with respect to what FluidSynth is doing). 

This linear workflow is always the most efficient in terms of minimizing overhead; the more you split the same amount of work up by threads, the greater amount of time is spent splitting and combining the results.

That works great if we can be sure atleast one single core is fast enough to keep our ring filled with rendered buffers. 

This should be the case on 'normal and modern' hardware, but I can  imagine a case where one core is never fast enough: say a tiny SBC operating at 200 MHz per core, but has 4 cores. 

Such a tiny device might need the synthesis spread across 3 threads to keep up while DOSBox operates on the 4th core. 

Another option would be to just set this at 2 internally (and drop the setting in the conf file), knowing that splitting the synth into two threads still isn't burning up much extra overhead, and should cover 99.99% of realistic user's hardware.. even folks older Rpi2's will be easily accommodated.

Reply:
*Sorry, I should've specified earlier. This ^^^^ is the only question I have about this PR.* :)

Reply:
@kcgen Hey, I am quite sure your comment did not exist last time I responded in this thread… Maybe some problem on GitHub side, whatever :)

> This linear workflow is always the most efficient in terms of minimizing overhead; (…)

Not always! **This PR** removes linear workflow to minimize overhead in main thread! Having FluidSynth do synthesis in a single thread *might be* appropriate, but *we don't know that yet* (!).

> the more you split the same amount of work up by threads, the greater amount of time is spent splitting and combining the results.

That's called algorithm scalability; *We don't know* how scalable internal FluidSynth algorithm really is. It might work ok, it might work not-so-ok, it might depend on user's hardware, it might depend on soundfont being used… 

Maybe we should indeed avoid exposing this setting to the end-users, but *we already do expose it*. I don't want to remove this setting just to bring it back in 0.78.0 if it'll turn out to be necessary for someone.

Can we bring this setting back, please? Maybe we'll remove it in the future (likely by replacing with some heuristic that will pick good value for user *somehow*), but I think we need it for now.